src/main/java/org/quiltmc/loader/impl/filesystem/QuiltJoinedFileSystemProvider.java

```diff
@@ -229,7 +229,7 @@ public DirectoryStream<Path> newDirectoryStream(Path dir, Filter<? super Path> f
 							return filter.accept(toJoinedPath(backing, path));
 						}));
 						anyReal = true;
-					} catch (NotDirectoryException e) {
+					} catch (NotDirectoryException | NoSuchFileException e) {
```

Comment:
Now that I'm looking at this it seems really odd to catch the exception - and not check `Files.isDirectory()` before invoking it?

I think the reason why this only caught `NotDirectoryException` before was because `Files.newDirectoryStream` names it explicitly - but I don't think it's a good idea to catch other exceptions too.